src/client/main.cpp

```diff
@@ -2,6 +2,87 @@
 #include <memory>
 #include <string>
 
+#include "helper/parser.h"
+
+using namespace std;
+
+string loggedInUsername = "";
```

Comment:
Try to avoid global variables. In this case it won't be much of an issue as there is no multithreading/several users on the client side, but if you add a reference to the functions that handle the username so you can modify it there or switch to a class-based approach that would look cleaner

Reply:
Changed.